tests/models/test_image_querying.py

```diff
@@ -121,121 +125,224 @@ def test_image_cone_search( provenance_base ):
-            soughtids = set( [ s.id for s in sought ] )
-            assert { image2.id }.issubset( soughtids )
-            assert len( { image1.id, image3.id, imagepoint.id, imagefar.id } & soughtids ) == 0
-
-            sought = Image.find_containing( imagepoint, session=session )
-            soughtids = set( [ s.id for s in sought ] )
-            assert { image1.id }.issubset( soughtids )
-            assert len( { image2.id, image3.id, imagepoint.id, imagefar.id } & soughtids ) == 0
-
-            sought = session.query( Image ).filter( Image.containing( 0, 0 ) ).all()
-            soughtids = set( [ s.id for s in sought ] )
-            assert len( { image1.id, image2.id, image3.id, imagepoint.id, imagefar.id } & soughtids ) == 0
+            session.add( image4 )
+            session.add( image5 )
+            # These tests don't pass if I don't commit here.  However, the tests above
+            # (in the for loop) did pass even though I didn't commit.  I don't understand
+            # the difference.  (Mumble mumble typical sqlalchmey mysteriousness mumble mumble.)
+            # In practical usage, we're going to be searching stuff that was committed to the
+            # database before, so the equivalent of this next commit will have been run.
+            session.commit()
 
-            sought = Image.find_containing( imagefar, session=session )
+            sought = Image.find_potential_overlapping( image1, session=session )
             soughtids = set( [ s.id for s in sought ] )
-            assert len( { image1.id, image2.id, image3.id, imagepoint.id, imagefar.id } & soughtids ) == 0
+            assert { image1.id, image2.id, image3.id, image4.id, image5.id }.issubset( soughtids )
 
-            sought = session.query( Image ).filter( Image.within( image1 ) ).all()
+            sought = Image.find_potential_overlapping( image2, session=session )
             soughtids = set( [ s.id for s in sought ] )
-            assert { image1.id, image2.id, image3.id, imagepoint.id }.issubset( soughtids )
-            assert len( { imagefar.id } & soughtids ) == 0
+            assert { image1.id, image2.id, image5.id }.issubset( soughtids )
+            assert len( { image3.id, image4.id } & soughtids ) == 0
 
-            sought = session.query( Image ).filter( Image.within( imagefar ) ).all()
+            sought = Image.find_potential_overlapping( image4, session=session )
             soughtids = set( [ s.id for s in sought ] )
-            assert len( { image1.id, image2.id, image3.id, imagepoint.id, imagefar.id } & soughtids ) == 0
+            assert { image1.id, image4.id, image5.id }.issubset( soughtids )
+            assert len( { image2.id, image3.id } & soughtids ) == 0
 
         finally:
-            session.rollback()
+            # When a test fails, this doesn't seem to actually
+            #  clean up the database.  Is this pytest subverting
+            #  the finally block?  Or is it some sqlqlchemy mysteriousness?
```

Comment:
could perhaps just try checking if it is in the session rather than persistent? Other than that, no clue...